tools/README.md

```diff
@@ -112,115 +77,199 @@ export repo_owner=<your github account>
 
-In order to directly get the password reset script, run the following:
+    These jobs are configured to use [Jenkinsfile](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile) and [Jenkinsfile.cicd](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile.cicd) found at the root of your project repository.  These Jenkinsfiles will make use of the OpenShift ConfigMaps we will create below.
+
+* **init.groovy.d/100-jenkins-config-set-admin-address.groovy**
+
+    This groovy script will update the admin email address.  It is not necessary, but an example of further customization to Jenkins.  If you are swiping this, keep in mind the email address is hardcoded.
+
+### Project Prerequisites
+
+We are setting up a CICD pipeline to build and deploy a specific project.  In order for this to work, there are expectations on those projects.  Currently, each project will require 3 files at the __root__ of the repository.
+
+| Name | Description |
+| --- | --- |
+| [Jenkinsfile](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile) | this is the master branch build pipeline |
+| [Jenkinsfile.cicd](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/Jenkinsfile.cicd) | this is the Pull Request build pipeline |
+| [sonar-project.properties](https://github.com/bcgov/nr-showcase-devops-tools/blob/master/sonar-project.properties) | this is will configure your SonarQube scans |
+
+This repository contains skeleton examples, see projects [NR Get Token](https://github.com/bcgov/nr-get-token.git) and [NR Messaging Service Showcase](https://github.com/bcgov/nr-messaging-service-showcase.git) for full examples.
+
+A very simple example to demonstrate a build and SonarQube analysis (no additional setup, no servers, no deployments), can be found at [nr-showcase-devops-tools-demo-sq](https://github.com/bcgov/nr-showcase-devops-tools-demo-sq.git).  This can be forked and used to test the Jenkins and SonarQube installation, and to demonstrate the master and pull request jobs.  For instance, the master branch will show Code Smells and duplicates in SonarQube.  In you fork, create a pull request that corrects these and see how the PR Jenkins job kicks off.
+
+### Jenkins Server Setup
+
+The following commands setup up Jenkins and uses this repository and specific OpenShift project namespaces.
+
+#### Create Secrets
+
+The BCDevOps CICD Jenkins Basic install requires a template github secret and a template for the slave.  This will create the secrets named as it requires.
 
 ```sh
-curl https://raw.githubusercontent.com/BCDevOps/sonarqube/bbb9f62e29706b61382cf24d7ad7e08f2476a01f/provisioning/updatesqadminpw.sh > updatesqadminpw.sh
+oc -n $tools process -f "$templates_url/secrets.yaml" -p GH_USERNAME=$gh_username -p GH_PASSWORD=$gh_password | oc  -n $tools create -f -
```

Comment:
Think we should just do the -l app=jenkins-prod and not update the templates.

Reply:
The app label is embedded into the template now